### PR TITLE
add: unit tests for the refactored code

### PR DESCRIPTION
**Pull request by gnhn**
- Created unit tests for refactored codebase
- Added `MainActivity2` for the refactored codebase

### CODE REVIEW

File: app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                        }
+                    }
+                    Type.VIDEO -> {
+                        if (SAVideoAd.hasAdAvailable(item.placementId)) {
+                            Log.i(TAG, "PLAYING VIDEO")
+                            SAVideoAd.play(item.placementId, this@MainActivity2)
+                        } else {
+                            SAVideoAd.load(item.placementId, this@MainActivity2)
+                        }
+//                        if (SAVideoAd.hasAdAvailable(item.placementId)) {
```
Comment (tashmulla) [inline app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:106]: Was this commented-out block meant to be left in here?

File: app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            Log.i(TAG, "Config 2 selected")
+            bannerView.disableParentalGate()
+            bannerView.disableBumperPage()
+
+            SAInterstitialAd.disableParentalGate()
+            SAInterstitialAd.disableBumperPage()
+
+            SAVideoAd.disableParentalGate()
+            SAVideoAd.disableBumperPage()
+            // SAVideoAd.enableCloseButtonNoDelay()
```
Comment (thomas-james-1986) [inline app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:57]: Should this be commented out here?

Comment (gnhn) [reply app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:57]: Yes, unfortunately the new code doesn't have that functionality. But, we can keep it commented out until it is added maybe?

File: app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt (hunk 1 above)
Comment (gnhn) [reply app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:106]: Yes, we don't have loading by lineItem, creativeId function at this moment. Missing feature on the refactor. Will be added and then we can uncomment.

Comment (thomas-james-1986) [inline app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:106]: Do we still need these commented out lines?

Comment (gnhn) [reply app/src/main/java/tv/superawesome/demoapp/MainActivity2.kt:106]: We can remove them and add them later when it is implemented, or just leave them for now

**Review by thomas-james-1986 — APPROVED** [review]

**Pull request merged**